lib/screens/v2/candidate_generator/widgets/alerts.ex

```diff
@@ -77,8 +77,9 @@ defmodule Screens.V2.CandidateGenerator.Widgets.Alerts do
     end
 
     alerts
-    |> Enum.filter(&(&1.effect in @relevant_effects))
-    |> Enum.filter(&Enum.any?(&1.informed_entities, relevant_ie?))
+    |> Stream.filter(&(&1.effect in @relevant_effects))
+    |> Stream.filter(&Enum.any?(&1.informed_entities, relevant_ie?))
+    |> Enum.to_list()
```

Comment:
Yes—Credo complains about repeated applications of `Enum.filter/2` and/or `Enum.reject/2` because they're less efficient than filtering an enumerable once with a single function that applies all of the checks simultaneously. Every time you call `Enum.filter/2` or `Enum.reject/2` on an enumerable, it iterates through the entire thing.

However, in some of these cases it seemed much more readable to split out the different conditions we were filtering the list by. E.g. here, we first select alerts whose effect we care about with `&(&1.effect in @relevant_effects)`. Then from those, we select the ones that inform an entity we care about (the screen's home stop, or a route that serves the home stop) with `&Enum.any?(&1.informed_entities, relevant_ie?)`.

So instead of making one function to do both checks at once, I used `Stream` functions. These apply lazily, so for example the second `Stream.filter` only acts on values that made it past the first one. We keep the readable code while also making Credo happy.
https://elixir-lang.org/getting-started/enumerables-and-streams.html#eager-vs-lazy
https://elixir-lang.org/getting-started/enumerables-and-streams.html#streams